### PR TITLE
[CI] bump action-ros-ci and pass ref

### LINKED ISSUE #619: CI on Humble Meta-Issue
**Describe the bug**
CI on Humble is pretty red right now.

Creating this Meta Issue to find and link any other Issue or PRs which might be related.

**To Reproduce**
image

**Expected behavior**
CI should be less red and more green

**Screenshots**
See Above

**Environment (please complete the following information):**
Github CI

**Additional context**
I'm preparing to release something that depends on ros2_controllers and I noticed the Source Build with ld errors failing joint_trajectory_controller

### PR DESCRIPTION
**Pull request by moriarty**
- Bump action-ros-ci to get option to pass in a ref
  - https://github.com/ros-tooling/action-ros-ci/issues/772
- fixes #619 where scheduled jobs ran with latest commit from default branch instead of the correct branch

### CODE REVIEW

**Review by aprotyas — APPROVED** [review]
Could be easier to just pin the action to `0.3`. I believe it picks the latest minor version anyway

**Review by destogl — APPROVED** [review]
Thanks!

File: .github/workflows/ci-coverage-build.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
         with:
           required-ros-distributions: ${{ env.ROS_DISTRO }}
       - uses: actions/checkout@v3
-      - uses: ros-tooling/action-ros-ci@0.3.0
+      - uses: ros-tooling/action-ros-ci@0.3.2
```
Comment (moriarty) [inline .github/workflows/ci-coverage-build.yml:23]:
```suggestion
      - uses: ros-tooling/action-ros-ci@0.3
```

File: .github/workflows/reusable-ros-tooling-source-build.yml [hunk 2]
```diff
@@ -32,9 +32,10 @@ jobs:
       - uses: actions/checkout@v3
         with:
           ref: ${{ inputs.ref }}
-      - uses: ros-tooling/action-ros-ci@0.3.0
+      - uses: ros-tooling/action-ros-ci@0.3.2
```
Comment (moriarty) [inline .github/workflows/reusable-ros-tooling-source-build.yml:35]:
```suggestion
      - uses: ros-tooling/action-ros-ci@0.3
```

**Review by moriarty — COMMENTED** [review]
Add suggested changes

**Comment by moriarty** [comment]
I think you always squash merge so I just used the GitHub browser ui to bring the branch up to date and apply the review suggestion

**Comment by moriarty** [comment]
> Could be easier to just pin the action to `0.3`. I believe it picks the latest minor version anyway

I thought so to... but I guess not
```
Error: Unable to resolve action `ros-tooling/action-ros-ci@0.3`, unable to find version `0.3`
```

**Pull request merged**